src/hub.rs

```diff
@@ -185,24 +148,55 @@ impl Hub {
                     }
-                }
-                Operation::SetMaterial(material) => if let SubNode::Visual(ref mut mat, _) = node.sub_node {
-                    *mat = material;
                 },
-                Operation::SetTexelRange(base, size) => if let SubNode::Visual(ref mut material, _) = node.sub_node {
-                    match *material {
-                        material::Material::Sprite(ref mut params) => params.map.set_texel_range(base, size),
-                        _ => panic!("Unsupported material for texel range request"),
+                Operation::AddChild(child_ptr) => match node.sub_node {
+                    SubNode::Group { ref mut first_child } => {
+                        let sibling = mem::replace(first_child, Some(child_ptr.clone()));
+                        deferred_sibling_updates.push((child_ptr, sibling));
                     }
+                    _ => unreachable!()
                 },
-                Operation::SetText(operation) => if let SubNode::UiText(ref mut data) = node.sub_node {
-                    Hub::process_text(operation, data);
+                Operation::SetAudio(operation) => match node.sub_node {
+                    SubNode::Audio(ref mut data) => {
+                        Hub::process_audio(operation, data);
+                    }
+                    _ => unreachable!()
                 },
-                Operation::SetShadow(map, proj) => if let SubNode::Light(ref mut data) = node.sub_node {
-                    data.shadow = Some((map, proj));
+                Operation::SetText(operation) => match node.sub_node {
+                    SubNode::UiText(ref mut data) => {
+                        Hub::process_text(operation, data);
+                    }
+                    _ => unreachable!()
+                },
+                Operation::SetShadow(map, proj) => match node.sub_node {
+                    SubNode::Light(ref mut data) => {
+                        data.shadow = Some((map, proj));
+                    }
+                    _ => unreachable!()
+                },
+                Operation::SetMaterial(material) => match node.sub_node {
+                    SubNode::Visual(ref mut mat, _) => {
+                        *mat = material;
+                    }
+                    _ => unreachable!()
+                },
+                Operation::SetTexelRange(base, size) => match node.sub_node {
+                    SubNode::Visual(Material::Sprite(ref mut params), _) => {
+                        params.map.set_texel_range(base, size);
+                    }
+                    _ => unreachable!()
                 },
+            };
+        }
+
+        for (child_ptr, sibling) in deferred_sibling_updates {
+            let child = &mut self.nodes[&child_ptr];
+            if child.next_sibling.is_some() {
+                error!("Attaching a child that still has an old parent, discarding siblings");
```

Comment:
How we can change parent now?

Reply:
Oh right, I forgot about this :) Now added to TODO list for this PR.
Basically, we can either merge the old siblings with the new ones (like `Scene::add` does), or require the user to call `Group::remove(&item)` before adding to another group. Either way, the `remove` needs to be added.